src/main/java/org/spdx/sbom/gradle/project/SpdxPkgInfo.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2023 The Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.spdx.sbom.gradle.project;
+
+import java.util.Optional;
+import org.immutables.serial.Serial;
+import org.immutables.value.Value.Immutable;
+import org.spdx.sbom.gradle.SpdxPackageExtension;
+
+@Immutable
+@Serial.Version(1)
+public interface SpdxPkgInfo {
+
+  String getName();
+
+  String getVersion();
+
+  boolean getCreatePackage();
+
+  Optional<ScmInfo> getScmInfo();
+
+  Optional<String> getSupplier();
+
+  static SpdxPkgInfo from(SpdxPackageExtension ext) {
+    if (ext == null) {
+      throw new NullPointerException("Extension value cannot be null");
+    }
+    var builder =
+        ImmutableSpdxPkgInfo.builder()
+            .name(ext.getName().get())
+            .version(ext.getVersion().get())
+            .createPackage(ext.getCreatePackage().get())
+            .supplier(Optional.ofNullable(ext.getSupplier().getOrNull()));
+
+    var scm = ext.getScm();
+    if (scm.getRevision().isPresent() && scm.getTool().isPresent() && scm.getUri().isPresent()) {
+      builder.scmInfo(
+          ScmInfo.from(scm.getTool().get(), scm.getUri().get(), scm.getRevision().get()));
+    }
+    if ((scm.getUri().isPresent() && !scm.getRevision().isPresent())
+        || (!scm.getUri().isPresent() && scm.getRevision().isPresent())) {
```

Comment:
If the tool is not present but the uri and revision are both present, I think this will skip line 50 but also not fail here. Should we add a check for `tool` here? Should we remove it from above?